test/jdk/javax/swing/JFileChooser/CustomApproveButtonTest.java

```diff
@@ -0,0 +1,99 @@
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+import java.awt.Component;
+import java.awt.Container;
+
+import javax.swing.JButton;
+import javax.swing.JFileChooser;
+import javax.swing.JFrame;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+
+/*
+ * @test
+ * @bug 8299522
+ * @key headful
+ * @summary JFileChooser Test to check for FileChooser's Approve Button Text
+ *          when CUSTOM_DIALOG type is set.
+ * @run main CustomApproveButtonTest
+ */
+
+public class CustomApproveButtonTest {
+    private JFrame frame;
+
+    public static void main(String[] args) throws Exception {
+        String aquaLAF = "com.apple.laf.AquaLookAndFeel";
+        SwingUtilities.invokeAndWait(new Runnable() {
+            public void run() {
+                UIManager.LookAndFeelInfo[] lookAndFeel = UIManager.getInstalledLookAndFeels();
+                for (UIManager.LookAndFeelInfo look : lookAndFeel) {
+                    if (look.getClassName().equals(aquaLAF)) {
+                        continue;
+                    }
+                    new CustomApproveButtonTest(look.getClassName());
+                }
+            }
+        });
+        System.out.println("Test Pass!");
+    }
+
+    private CustomApproveButtonTest(String lookAndFeel) {
+        System.out.println("Testing Look & Feel : " + lookAndFeel);
+        frame = new JFrame("CustomApproveButtonTest");
+
+        try {
+            UIManager.setLookAndFeel(lookAndFeel);
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to set " + lookAndFeel +
+                    " LookAndFeel. Error : " + e);
+        }
+
+        JFileChooser fileChooser = new JFileChooser();
+        fileChooser.setDialogType(JFileChooser.CUSTOM_DIALOG);
+        frame.add(fileChooser);
+        frame.setVisible(true);
+        frame.pack();
+
+        JButton customApproveButton = fileChooser.getUI().getDefaultButton(fileChooser);
+
+        if (customApproveButton == null) {
+            fail("Cannot find Custom Approve Button in FileChooser!");
+            return;
+        }
+        if (customApproveButton.getText() == null) {
```

Comment:
Since in aqua the fallback text is handled already and in aqua there is a sub panel which handles the Approve button, I excluded it.

Reply:
Also, `getDefaultApproveButton()` returns null for Aqua L&F, to include Aqua in the test we can use `getApproveButtonText()`.

Reply:
Here you are.

[JDK-8300084](https://bugs.openjdk.org/browse/JDK-8300084): AquaFileChooserUI.getDefaultButton returns null

Now you can fix this bug so that Aqua L&F doesn't return null and add that bugid to this test.